### PR TITLE
Improve cyclic dependency detection and err output

### PR DESCRIPTION
**Pull request by dankwart-de**
Removed an instruction from RoddyCLIClient.outputtestrunresult. This
line converted the configuration to what? So I got rid of it. The line
showed exceptions with cyclic dependencies.

Configuration.getID() will return "Unnamed configuration", if no
preloadedConfiguration is set.

Moved ConfigurationValue.java => ConfigurationValue.groovy
- Escape all kinds of $ to \$ otherwise replacements won't work.
- Instead of a RuntimeException, throw a ConfigurationError now when it
  comes to cyclic dependencies

In Aalysis class, upon run, catch ConfigurationError as more some type
of a known error. The developer needs to add ConfigurationError
instances to the configuration so they get printed!

Let ExecutionContext.execute throw a ConfigurationError

SimpleCohortWorkflow now contains a small cyclic dependency.

### CODE REVIEW

File: RoddyCore/src/de/dkfz/roddy/config/Configuration.java [hunk 1]
```diff
@@ -201,7 +201,10 @@ public String getName() {
      */
     @Override
     public String getID() {
-        return preloadedConfiguration.id;
+        if(preloadedConfiguration != null)
+            return preloadedConfiguration.id;
+        else
+            return "'Unnamed Configuration'";
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/config/Configuration.java]: Why are there single ticks inside the double ticks?

File: RoddyCore/src/de/dkfz/roddy/config/ConfigurationValue.groovy [hunk 2]
```diff
@@ -119,7 +121,7 @@ private String checkAndCorrectPathThrow(String temp) throws ConfigurationError {
         String curUserPath = (new File("")).getAbsolutePath();
         String applicationDirectory = Roddy.getApplicationDirectory().getAbsolutePath();
         //TODO Make something like a blacklist. This is not properly handled now. Initially this was done because Java sometimes puts something in front of the file paths.
-        if (value.startsWith("${") || value.startsWith("$") || value.startsWith("~") || !value.startsWith("/")) {
+        if (value.startsWith('${') || value.startsWith('$') || value.startsWith("~") || !value.startsWith("/")) {
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/config/ConfigurationValue.groovy]: Now that you touched that code, would it not make sense generalizing it and moving it into e.g. a dedicated class. From there we could evolve it into a true templating system, or use an existing templating system. We use similar code also for the variables in the filename patterns. Also, there are plenty of repetitions here -- '${', etc.

File: RoddyCore/src/de/dkfz/roddy/core/Analysis.java [hunk 3]
```diff
@@ -448,6 +451,9 @@ protected void executeRun(ExecutionContext context, boolean preventLoggingOnQuer
                     }
                 }
             }
+        } catch (ConfigurationError ce) {
```
Comment (vinjana) [inline RoddyCore/src/de/dkfz/roddy/core/Analysis.java]: What? Actually, I also touched that code and more.

File: RoddyCore/src/de/dkfz/roddy/config/Configuration.java (hunk 1 above)
Comment (dankwart-de) [reply RoddyCore/src/de/dkfz/roddy/config/Configuration.java]: For readability.

**Comment by dankwart-de** [comment]
This PR will be blocked. vinjana has a local branch with a lot changes at the same place! Leave for now.
Related to #79

File: RoddyCore/src/de/dkfz/roddy/config/Configuration.java (hunk 1 above)
Comment (vinjana) [reply RoddyCore/src/de/dkfz/roddy/config/Configuration.java]: They don't belong there. You need to add them to the place, where the string is printed.

File: RoddyCore/src/de/dkfz/roddy/config/ConfigurationValue.groovy (hunk 2 above)
Comment (dankwart-de) [reply RoddyCore/src/de/dkfz/roddy/config/ConfigurationValue.groovy]: It would. But not for now, that'll be a bigger discussion right?

Comment (vinjana) [reply RoddyCore/src/de/dkfz/roddy/config/ConfigurationValue.groovy]: Right.

**Comment by dankwart-de** [comment]
Discussed with vinjana, will merge

**Pull request merged**